src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestConfigureCallGen.generated.txt

```diff
@@ -0,0 +1,91 @@
+#nullable enable
+
+using System.Linq;
+
+internal static class GeneratedConfigurationBinder
+{
+    public static global::Microsoft.Extensions.DependencyInjection.IServiceCollection Configure<T>(this global::Microsoft.Extensions.DependencyInjection.IServiceCollection services, global::Microsoft.Extensions.Configuration.IConfiguration configuration)
+    {
+        if (typeof(T) == typeof(Program.MyClass))
+        {
+            return services.Configure<Program.MyClass>(obj =>
+            {
+                BindCore(configuration, ref obj);
+            });
+        }
+
+        throw new global::System.NotSupportedException($"Unable to bind to type '{typeof(T)}': 'Generator parser did not detect the type as input'");
+    }
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+
+        if (configuration["MyString"] is string stringValue1)
+        {
+            obj.MyString = stringValue1;
+        }
+
+        if (configuration["MyInt"] is string stringValue2)
+        {
+            obj.MyInt = int.Parse(stringValue2);
+        }
+
+        System.Collections.Generic.List<int> temp3 = obj.MyList;
+        temp3 ??= new System.Collections.Generic.List<int>();
+        BindCore(configuration.GetSection("MyList"), ref temp3);
+        obj.MyList = temp3;
+
+        System.Collections.Generic.Dictionary<string, string> temp4 = obj.MyDictionary;
+        temp4 ??= new System.Collections.Generic.Dictionary<string, string>();
+        BindCore(configuration.GetSection("MyDictionary"), ref temp4);
+        obj.MyDictionary = temp4;
+
+    }
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.List<int> obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+
+        int element;
+        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
+        {
+            if (section.Value is string stringValue5)
+            {
+                element = int.Parse(stringValue5);
+                obj.Add(element);
+            }
+        }
+    }
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.Dictionary<string, string> obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+
+        string key;
+        foreach (Microsoft.Extensions.Configuration.IConfigurationSection section in configuration.GetChildren())
+        {
+            if (section.Key is string stringValue6)
+            {
+                key = stringValue6;
+                string element;
```

Comment:
Why is this extra variable needed?

Reply:
This is leveraging a more general implementation pattern for parsing string values. I'll clean up and simplify.